### DISCUSSION TITLE
Reactive queryKey with vue-query?

### DISCUSSION THREAD

**Discussion by geekox86** [TanStack/query#4332] [Q&A] [answered]
Greetings,

I have a queryKey like this:
["example", props.id, someVueRefOrComputed]

How can I make the queryKey change when the props.id or my ref/computed changes which should trigger the query to reload? What are the other options?

Also, how can I access the key from a mutation function?

Many thx!

**Comment by DamianOsipiuk** [comment, marked as answer]
All `ref` and `computed` provided as a part of `queryKey` or `options` will be automatically observed and changes would be reflected with the Observer.

Query key can be accessed from the `queryFn` context: https://vue-query-next.vercel.app/#/guides/query-functions?id=query-function-variables
